ui/src/components/forms/SegmentsPicker.tsx

```diff
@@ -68,43 +79,70 @@ export default function SegmentsPicker(props: SegmentPickerProps) {
                   filterValue: truncateKey(s.key),
                   displayValue: s.name
                 }))}
+              disabled={readonly}
               selected={selectedSegment}
               setSelected={(filterableSegment) => {
                 handleSegmentSelected(index, filterableSegment);
               }}
+              inputClassNames={
+                readonly
+                  ? 'cursor-not-allowed bg-gray-100 text-gray-500'
+                  : undefined
+              }
             />
           </div>
-          <div>
-            <button
-              type="button"
-              className="text-gray-400 mt-2 hover:text-gray-500"
-              onClick={() => handleSegmentRemove(index)}
-            >
-              <MinusSmallIcon className="h-6 w-6" aria-hidden="true" />
-            </button>
-          </div>
+          {editing && parentSegments.length - 1 === index ? (
+            <div>
+              <button
+                type="button"
+                className={twMerge(`
+                  text-gray-400 mt-2 hover:text-gray-500 ${
+                    readonly ? 'hover:text-gray-400' : undefined
+                  }`)}
+                onClick={() => setEditing(false)}
+                title={readonly ? 'Not allowed in Read-Only mode' : undefined}
+                disabled={readonly}
+              >
+                <PlusSmallIcon className="h-6 w-6" aria-hidden="true" />
+              </button>
+            </div>
+          ) : (
+            <div>
+              <button
+                type="button"
+                className="text-gray-400 mt-2 hover:text-gray-500"
+                onClick={() => handleSegmentRemove(index)}
+                title={readonly ? 'Not allowed in Read-Only mode' : undefined}
+                disabled={readonly}
+              >
+                <MinusSmallIcon className="h-6 w-6" aria-hidden="true" />
+              </button>
+            </div>
+          )}
         </div>
       ))}
-      <div className="w-full">
-        <div className="w-5/6">
-          <Combobox<FilterableSegment>
-            id={`segmentKey-${parentSegments.length}`}
-            name={`segmentKey-${parentSegments.length}`}
-            placeholder="Select or search for a segment"
-            values={segments
-              .filter((s) => !segmentsSet.current!.has(s.key))
-              .map((s) => ({
-                ...s,
-                filterValue: truncateKey(s.key),
-                displayValue: s.name
-              }))}
-            selected={null}
-            setSelected={(filterableSegment) => {
-              handleSegmentSelected(parentSegments.length, filterableSegment);
-            }}
-          />
+      {(!editing || parentSegments.length === 0) && (
+        <div className="w-full">
+          <div className="w-5/6">
+            <Combobox<FilterableSegment>
```

Comment:
should we set `disabled=${readOnly}` here?

Reply:
Do you mean like this? https://github.com/flipt-io/flipt/blob/0eb7379823e7cc173024890023226acc8fcf96e1/ui/src/components/forms/SegmentsPicker.tsx#L82

Or something else?

Reply:
yah, nvm i see that the above is only rendered when in 'editing' mode